03/Category.js

```diff
@@ -1,7 +1,48 @@
-import React from 'react';
+import React from "react";
+import Product from "./Product";
 
-function Category(props) {
-    return <h2>Category</h2>
+const styles = {
+    border: "1px solid black",
+    width: "100%",
+    padding: "5px",
+};
+
+const productCont = {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "space-between",
+};
+
+class Category extends React.Component {
+    handleAddToCart = (id) => {
+        console.log("adding to cart");
+        this.props.addToCart(id);
+    };
+
+    render() {
+        const { data } = this.props;
+        const productsList = data.map((p) => {
+            return (
+                <div style={productCont} key={p.id}>
+                    <Product name={p.name} price={p.price} id={p.id} />{" "}
+                    <button
+                        style={{ height: "20px" }}
+                        onClick={() => {
+                            this.handleAddToCart(p);
+                        }}
+                    >
+                        Add to cart
+                    </button>
+                </div>
```

Comment:
Zobacz, że ta część jest identyczna z tym co mamy w `<Cart>`.
Można by było zrobić tak: `<Product name={p.name} price={p.price} id={p.id} isCart={false} handleButtonClick={() => this.handleAddToCart(p.id)}/>`, a w samym `<Product/>` zrobić coś takiego:
```
                <div style={productCont} key={id}>
                    <div>
                        <p>
                            {name}: {price}
                        </p>
                    </div>
                    <button
                        style={{ height: "20px" }}
                        onClick={() => {
                            handleButtonClick(p);
                        }}
                    >
                        {isCart ? 'Delete' : 'Add to cart'}
                    </button>
                </div>
```

Ale to co zrobiłeś też jest ok - chciałem pokazać po prostu alternatywę ;)